website/docs/advanced-concepts/access-control/roles.md

```diff
@@ -0,0 +1,237 @@
+
+##### Workspace 
+
+ If you give edit permission to the workspace, it gives access to modify all the applications in the respective workspace.
+
+:::info
+Giving `Edit` permission to a workspace has a cascading effect, i.e, it gives edit and view permission to all the applications in the workspace. 
+:::
+
+##### Application
+
+If you want the user to have edit access only to a specific application in a workspace, open the respective workspace's dropdown and check the edit permission adjacent to the respective application. `Edit` for just an application allows you to edit resources like page, actions (APIs, queries, JSObjects), Git connection etc. 
+
+##### Page
+
+`Edit` access for a particular page allows you to make changes on the respective page. You can edit the page name, clone the page, set it as the home page, and toggle their visibility.
+
+#### View 
+
+`View` only allows a user to just have view access for the existing applications, pages, or actions (APIs, queries, JSObjects) in a workspace. You can't make any changes if you only have the View permission. It applies to all the hierarchical levels in a workspace.
+
+#### Delete
+
+`Delete` gives the user an access to delete the existing applications, pages, or actions (APIs, queries, JSObjects) in a workspace. With the delete permission, one can't edit or create resources, but, the user has the access to delete it.
+
+Delete permission behaves differently at various hierarchical levels -
+
+##### Workspace 
+
+ If you give delete permission to the workspace, It gives access to remove any application from the respective workspace. 
+
+##### Application
+
+If you want the user to have delete access only to a specific application in a workspace, open the respective workspace's dropdown and check the delete permission adjacent to the respective application. `Delete` for just an application allows you to delete resources like page, actions (APIs, queries, JSObjects).
+
+##### Page
+
+`Delete` access for a particular page allows you to delete the respective page or the resources in the page like APIs, queries, JSObjects. 
+
+#### Export
+
+`Export` allows a user to export the applications in a workspace. If a user has export permission enabled, he can also view the applications in the workspace.
+This permission can also be enabled for a particular application, allowing the user to export and view the respective application only.
+
+#### Make public
+
+`Make Public` gives the user an ability to make the application public or private in a workspace.
+
+### Datasource and queries permissions
+
+For datasource and queries, here are the permissions that can be assigned to user or a group -
+
+
+#### Execute
+
+`Execute` gives a user the ability to execute the datasource. This permission can be given for a workspace or a particular datasource in the workspace.
+
+##### Workspace
+
+Giving execute permission at a workspace level allows the user to execute any datasource in the workspace.
```

Comment:
its not permission to execute a datasource. It is permission to execute queries in a datasource

Reply:
That's the catch here. It gives you access to execute the datasource. For access to execute queries, you have to go to the page level and give execute access to the specific query.
Execute permission at a workspace level is required for a query to work, but you must also enable execute access on the query itself.